From 18f4503774bdccf3075c0e4f8f4e873b59f09ec0 Mon Sep 17 00:00:00 2001
From: Jorge QUINTERO <jorge.quintero@intel.com>
Date: Wed, 26 Sep 2012 17:35:47 +0200
Subject: [PORT FROM MAIN] VoIP cuts in UL

BZ: 57086

Change coming from R3 (BZ 58342)

Feature: AudioComms, VoIP
Category: Integration
Purpose: PV-Enabler
Origin: Internal
Upstream-candidate: yes
Origin-Change-Id: I6a1d6a185b3cbb8d28102a81cdaae879eba1f23c
Change-Id: I95be746ba1b8c9915be8e371395f8e7d4dc1ac0f
Orig-Change-Id: I85bedd735b60510f9dbe653937cdb863e0fdc38c
Reviewed-on: http://android.intel.com:8080/76842
Reviewed-by: Barthes, FabienX <fabienx.barthes@intel.com>
Tested-by: Barthes, FabienX <fabienx.barthes@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 services/audioflinger/Threads.cpp |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/services/audioflinger/Threads.cpp b/services/audioflinger/Threads.cpp
index 47ca100..5590928 100644
--- a/services/audioflinger/Threads.cpp
+++ b/services/audioflinger/Threads.cpp
@@ -2964,6 +2964,14 @@ bool AudioFlinger::MixerThread::checkForNewParameters_l()
                 reconfig = true;
             }
         }
+        if (param.getInt(String8(AudioParameter::keyInputSource), value) == NO_ERROR) {
+            if (value == AUDIO_SOURCE_VOICE_COMMUNICATION) {
+                //
+                // Might have different buffering model applied for VoIP input source
+                //
+                reconfig = true;
+            }
+        }
         if (param.getInt(String8(AudioParameter::keyRouting), value) == NO_ERROR) {
 #ifdef ADD_BATTERY_DATA
             // when changing the audio output device, call addBatteryData to notify
-- 
1.7.4.1

